sklearnex/_device_offload.py

```diff
@@ -0,0 +1,108 @@
+#===============================================================================
```

Comment:
I am a little confused that the auxiliary functionality lies at the root of the project. Can I put them in a utils folder?